perses/tests/test_hybrid_builder.py

```diff
@@ -242,16 +242,90 @@ def compute_logP(system, positions, parameter=None):
-    hybrid = HybridTopologyFactory(leucine_system, alanine_system, leucine_topology, alanine_topology, leucine_positions, alanine_positions, atom_map, softening=0.0)
+    #hybrid = HybridTopologyFactory(leucine_system, alanine_system, leucine_topology, alanine_topology, leucine_positions, alanine_positions, atom_map, softening=0.0)
+    top_prop = TopologyProposal(new_topology=leucine_topology, new_system=leucine_system, old_topology=alanine_topology, old_system=alanine_system, new_to_old_atom_map=atom_map)
+    hybrid_factory = new_relative.HybridTopologyFactory(top_prop)
+
     [system, topology, positions, sys2_indices_in_system, sys1_indices_in_system] = hybrid.createPerturbedSystem()
 
-    compute_alchemical_correction(leucine_system, alanine_system, system, leucine_positions, positions, positions, alanine_positions)
+    compute_alchemical_correction(leucine_system, alanine_system, hybrid_new, leucine_positions, positions, positions, alanine_positions)
+
+def compare_energies(mol_name="naphthalene", ref_mol_name="benzene"):
+    """
+    Make an atom map where the molecule at either lambda endpoint is identical, and check that the energies are also the same.
+    """
+    from topology_proposal import SmallMoleculeSetProposalEngine, TopologyProposal
+    import openeye.oechem as oechem
+    from new_relative import HybridTopologyFactory
+    import simtk.openmm as openmm
+
+    from perses.tests.utils import createOEMolFromIUPAC, createSystemFromIUPAC
+
+    mol_name = "naphthalene"
+    ref_mol_name = "benzene"
+
+    mol = createOEMolFromIUPAC(mol_name)
+    m, system, positions, topology = createSystemFromIUPAC(mol_name)
+
+    refmol = createOEMolFromIUPAC(ref_mol_name)
+
+    #map one of the rings
+    atom_map = SmallMoleculeSetProposalEngine._get_mol_atom_map(mol, refmol)
+
+    #now use the mapped atoms to generate a new and old system with identical atoms mapped. This will result in the
+    #same molecule with the same positions for lambda=0 and 1, and ensures a contiguous atom map
+    effective_atom_map = {value : value for value in atom_map.values()}
+
+    #make a topology proposal with the appropriate data:
+    top_proposal = TopologyProposal(new_topology=topology, new_system=system, old_topology=topology, old_system=system, new_to_old_atom_map=effective_atom_map, new_chemical_state_key="n1", old_chemical_state_key='n2')
+
+    factory = HybridTopologyFactory(top_proposal, positions, positions)
+
+    alchemical_system = factory.hybrid_system
+    alchemical_positions = factory.hybrid_positions
+
+    integrator = openmm.VerletIntegrator(1)
+    platform = openmm.Platform.getPlatformByName("Reference")
+    context = openmm.Context(alchemical_system, integrator, platform)
+
+    context.setPositions(alchemical_positions)
+
+    functions = {
+        'lambda_sterics' : '2*lambda * step(0.5 - lambda) + (1.0 - step(0.5 - lambda))',
+        'lambda_electrostatics' : '2*(lambda - 0.5) * step(lambda - 0.5)',
```

Comment:
Ah, I see---these particular functions may not matter if you're just checking lambda 0 -> 1.